store/postgres/src/store.rs

```diff
@@ -815,37 +815,13 @@ impl StoreTrait for Store {
             })
     }
 
-    fn set_block_ptr_with_no_changes(
-        &self,
-        subgraph_id: SubgraphDeploymentId,
-        block_ptr_from: EthereumBlockPointer,
-        block_ptr_to: EthereumBlockPointer,
-    ) -> Result<bool, StoreError> {
-        let ops = SubgraphDeploymentEntity::update_ethereum_block_pointer_operations(
-            &subgraph_id,
-            block_ptr_from,
-            block_ptr_to,
-            "skip blocks with no changes",
-        );
-        let conn = self.get_entity_conn(&subgraph_id).map_err(Error::from)?;
-        let event = conn.transaction(|| self.apply_metadata_operations_with_conn(&conn, ops))?;
-
-        // Send the event separately, because NOTIFY uses a global DB lock.
-        conn.transaction(|| conn.send_store_event(&event))?;
-        conn.should_migrate(&subgraph_id, &block_ptr_to)
-    }
-
     fn transact_block_operations(
         &self,
         subgraph_id: SubgraphDeploymentId,
-        block_ptr_from: EthereumBlockPointer,
         block_ptr_to: EthereumBlockPointer,
         mods: Vec<EntityModification>,
     ) -> Result<bool, StoreError> {
-        // Sanity check on block numbers
-        if block_ptr_from.number != block_ptr_to.number - 1 {
-            panic!("transact_block_operations must transact a single block only");
-        }
+        let block_ptr_from = self.block_ptr(subgraph_id.clone())?;
```

Comment:
I am all for getting rid of `set_block_ptr_with_no_changes`, but I don't like that this change causes a query for data the caller already has. We should continue to pass `block_ptr_from` into `transact_block_operations` but change the rules for it: if `block_ptr_from + 1 < block_ptr_to`, `mods` has to be empty, and we must always have `block_ptr_from < block_ptr_to`.

Reply:
It's interesting observation that at some earlier point the block stream called made this same DB lookup, and we could plumb that through in order to save this call. But if we wanted to keep `block_ptr` cached in memory for performance, I'd approach that differently by having the store keep that cache without complicating other code. However this query should be cheap enough to do in both places, if it's currently slow that's because we're pulling the entire subgraph only to read the block pointer, which is an issue regardless of this change.

I can assert that `block_ptr_from < block_ptr_to`, though I don't follow why if `block_ptr_from + 1 < block_ptr_to` then `mods` must be empty, that's not the case in this PR, is that a problem?

Reply:
I measured the speed of this query for the Moloch subgraph, it never took more than 1 ms, so we're good on that.

Reply:
What I was getting at is that current callers have `block_ptr_from` already, so why not continue to pass that in rather than look it up? Yes, it's a very cheap lookup, at least on an empty, uncontested DB, but I'd rather not query at all when it's not necessary.

As for empty `mods`, I was going by what the current `set_block_ptr_with_no_changes` does; what is in `mods` when we are jumping more than one block? I'd like to assert/check _something_ about them; part of the reason for this is #1093 where we never got to the bottom of why we trip over a metadata update, even though that should be impossible, and I'd like to make sure we are not just silently overwriting stuff.

Reply:
The check we had `transact_block_operations` caught bugs in my early block explorer data work, so I second the request for keeping the from pointer and performing the sanity checks that @lutter is recommending.

Reply:
As am I (fine with the changes, that is).